src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -259,7 +259,7 @@ function assertValidProps(component, props) {
     );
     invariant(
       typeof props.dangerouslySetInnerHTML === 'object' &&
-      '__html' in props.dangerouslySetInnerHTML,
+      keyOf({ __html: null }) in props.dangerouslySetInnerHTML,
```

Comment:
Like this please: https://github.com/facebook/react/blob/401e6f10587b09d4e725763984957cf309dfdc30/src/isomorphic/deprecated/cloneWithProps.js#L21